lib/xnvme_be_spdk_dev.c

```diff
@@ -26,63 +26,61 @@
 static struct xnvme_be_spdk_ctrlr_ref g_cref[XNVME_BE_SPDK_CREFS_LEN];
 
 /**
- * look for a cref for the given given ident
+ * Increment the refcount for the ctrlr with the given ident. If ctrlr is NULL, only lookup based
+ * on ident.
  *
- * @return On success, a pointer to the cref is returned. When none exists, NULL is returned.
+ * @return On success, a pointer to the ctrlr is returned. On error, NULL is returned.
  */
 static struct spdk_nvme_ctrlr *
-_cref_lookup(struct xnvme_ident *id)
+_cref_ref(struct xnvme_ident *ident, struct spdk_nvme_ctrlr *ctrlr)
 {
+	int free_pos = -1;
 	for (int i = 0; i < XNVME_BE_SPDK_CREFS_LEN; ++i) {
-		if (strncmp(g_cref[i].uri, id->uri, XNVME_IDENT_URI_LEN - 1)) {
+		if (!g_cref[i].ctrlr) {
+			if (free_pos < 0 && !g_cref[i].ctrlr) {
```

Comment:
```suggestion
			if (free_pos < 0) {
```
`!g_cref[i].ctrlr` will always be true if we hit this path